### DISCUSSION TITLE
Logback: Make the logging asynchronous

### DISCUSSION THREAD

**Discussion by quantranhong1999** [linagora/tmail-backend#1136] [Ideas]
![image](https://github.com/user-attachments/assets/6b63debe-37df-4719-abe1-7a7586e3fb07)

Should we make the logging async to avoid blocking the app thread? I/O to disk for persisting log has more or less its cost IMO.

**Comment by chibenwa** [comment]
This would be awesome!

**Comment by Arsnael** [comment]
+1 :) Maybe just fire a mail in the ML regarding this @quantranhong1999 ?

**Reply by quantranhong1999** [reply]
Hmm this one is just configuration in `logback.xml`. I am not sure if we should tell people to adapt their configuration. Maybe we can just apply it on tmail-backend.

**Reply by Arsnael** [reply]
Ah yeah fair :)

**Reply by chibenwa** [reply]
Still can be beneficial and worth sharing with community and this could become a default conf, or a commented block into the default conf.

**Comment by vttranlina** [comment]
new knowledge, thank you
